source/common/log_sink/conveyor.h

```diff
@@ -0,0 +1,149 @@
+#pragma once
+
+#include <condition_variable>
+#include <deque>
+#include <memory>
+#include <utility>
+#include <vector>
+
+namespace Envoy {
+namespace LogTransport {
```

Comment:
Agree, log_sink was the name I came up trying to be general about log+stats but I'd like a more general term... thinking, suggestions welcome.